### PR TITLE
3 3 lwm2m service

### PR DESCRIPTION
**Pull request by nickAS21**
3/3_lwm2m_service

### CODE REVIEW

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2MTransportRequest.java [hunk 1]
```diff
@@ -237,45 +230,37 @@ else if (!contentFormat.equals(ContentFormat.TLV) && !resource.multiple) {
...
                         ((Response) response.getCoapResponse()).getCode(), response.getCode().getCode(), response.getCode().getName(), request.getPath().toString());
                 service.sentLogsToThingsboard(msg, registration);
                 log.error("[{}] - [{}] [{}] error SendRequest", ((Response) response.getCoapResponse()).getCode(), response.getCode(), request.getPath().toString());
-                if (request instanceof WriteRequest && ((WriteRequest) request).isReplaceRequest() && isDelayedUpdate) {
-                    this.handleResponseError(registration, request.getPath().toString(), lwM2MClient, isDelayedUpdate);
-                }
             }
-
         }, e -> {
+            if (!lwM2MClient.isInit()) lwM2MClient.initValue(this.service, request.getPath().toString());
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2MTransportRequest.java:257]: add braces for "if"

File: ui-ngx/src/app/modules/home/components/profile/device/lwm2m/lwm2m-device-profile-transport-configuration.component.html [hunk 2]
```diff
@@ -20,7 +20,24 @@
...
       <ng-template matTabContent>
         <section [formGroup]="lwm2mDeviceProfileFormGroup">
-          <div class="mat-padding">
+          <div class="mat-padding" style="padding-bottom: 0px">
+            <mat-checkbox formControlName="clientOnlyObserveAfterConnect" color="primary"
+                          matTooltip="{{ translate.get('device-profile.lwm2m.client-only-observe-after-connect-tip',
+                          { count: +lwm2mDeviceProfileFormGroup.get('clientOnlyObserveAfterConnect').value }) | async }}"
+                          matTooltipPosition="above">
+              {{ translate.get('device-profile.lwm2m.client-only-observe-after-connect',
+              { count: +lwm2mDeviceProfileFormGroup.get('clientOnlyObserveAfterConnect').value }) | async }}
```
Comment (vvlladd28) [inline ui-ngx/src/app/modules/home/components/profile/device/lwm2m/lwm2m-device-profile-transport-configuration.component.html:29]: Your need to used the translation pipe with params

File: ui-ngx/src/app/modules/home/components/profile/device/lwm2m/lwm2m-device-profile-transport-configuration.component.html [hunk 3]
```diff
@@ -20,7 +20,24 @@
...
+            </mat-checkbox>
+            <mat-checkbox *ngIf="!lwm2mDeviceProfileFormGroup.get('clientOnlyObserveAfterConnect').value"
+                          formControlName="clientUpdateValueAfterConnect" color="primary"
+                          matTooltip="{{ translate.get('device-profile.lwm2m.client-update-value-after-connect-tip',
+                          { count: +lwm2mDeviceProfileFormGroup.get('clientUpdateValueAfterConnect').value }) | async }}"
+                          matTooltipPosition="above">
+              {{ translate.get('device-profile.lwm2m.client-update-value-after-connect',
+              { count: +lwm2mDeviceProfileFormGroup.get('clientUpdateValueAfterConnect').value }) | async }}
```
Comment (vvlladd28) [inline ui-ngx/src/app/modules/home/components/profile/device/lwm2m/lwm2m-device-profile-transport-configuration.component.html:37]: Your need to used the translation pipe with params

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2MTransportRequest.java [hunk 4]
```diff
@@ -237,45 +230,41 @@ else if (!contentFormat.equals(ContentFormat.TLV) && !resource.multiple) {
...
-                this.handleResponse(registration, request.getPath().toString(), response, request, lwM2MClient, isDelayedUpdate);
+                this.handleResponse(registration, request.getPath().toString(), response, request);
                 if (request instanceof WriteRequest && ((WriteRequest) request).isReplaceRequest()) {
-                    String delayedUpdateStr = "";
-                    if (isDelayedUpdate) {
-                        delayedUpdateStr = " (delayedUpdate) ";
-                    }
-                    String msg = String.format(LOG_LW2M_INFO + ": sendRequest Replace%s: CoapCde - %s Lwm2m code - %d name - %s Resource path - %s value - %s SendRequest to Client",
-                            delayedUpdateStr, ((Response) response.getCoapResponse()).getCode(), response.getCode().getCode(), response.getCode().getName(), request.getPath().toString(),
+                    String msg = String.format(LOG_LW2M_INFO + ": sendRequest Replace: CoapCde - %s Lwm2m code - %d name - %s Resource path - %s value - %s SendRequest to Client",
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2MTransportRequest.java:245]: add LOG_LW2M_INFO use %s

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2MTransportRequest.java [hunk 5]
```diff
@@ -237,45 +230,41 @@ else if (!contentFormat.equals(ContentFormat.TLV) && !resource.multiple) {
...
+                            ((Response) response.getCoapResponse()).getCode(), response.getCode().getCode(), response.getCode().getName(), request.getPath().toString(),
                             ((LwM2mSingleResource) ((WriteRequest) request).getNode()).getValue().toString());
                     service.sentLogsToThingsboard(msg, registration);
-                    log.info("[{}] - [{}] [{}] [{}] Update SendRequest[{}]", ((Response) response.getCoapResponse()).getCode(), response.getCode(), request.getPath().toString(),
-                            ((LwM2mSingleResource) ((WriteRequest) request).getNode()).getValue(), delayedUpdateStr);
+                    log.info("[{}] [{}] - [{}] [{}] Update SendRequest[{}]", registration.getEndpoint(), ((Response) response.getCoapResponse()).getCode(), response.getCode(), request.getPath().toString(),
+                            ((LwM2mSingleResource) ((WriteRequest) request).getNode()).getValue());
                 }
             } else {
                 String msg = String.format(LOG_LW2M_ERROR + ": sendRequest: CoapCode - %s Lwm2m code - %d name - %s Resource path - %s  SendRequest to Client",
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2MTransportRequest.java:253]: add LOG_LW2M_INFO use %s

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2MTransportRequest.java [hunk 6]
```diff
@@ -308,54 +297,27 @@ private WriteRequest getWriteRequestSingleResource(ContentFormat contentFormat,
...
      * processing a response from a client
      * @param registration -
      * @param path -
      * @param response -
-     * @param lwM2MClient -
      */
-    private void sendResponse(Registration registration, String path, LwM2mResponse response, DownlinkRequest request, LwM2MClient lwM2MClient, boolean isDelayedUpdate) {
-        if (response instanceof ObserveResponse) {
+    private void sendResponse(Registration registration, String path, LwM2mResponse response, DownlinkRequest request) {
+        if (response instanceof ObserveResponse || response instanceof ReadResponse) {
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/server/LwM2MTransportRequest.java:317]: check

**Pull request merged**